SSD1306 OLED Cookbook

erazor666 opened the pull request:
## Description:


**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>
**Pull request in [esphome-core](https://github.com/esphome/esphome-core) with C++ framework changes (if applicable):** esphome/esphome-core#<esphome-core PR number goes here>

## Checklist:

  - [ ] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.

erazor666 committed:
Initial commit

erazor666 committed:
Initial commit

erazor666 committed:
Initial commit

erazor666 committed:
Update index.rst

erazor666 commented:
Re based on current and fixed white spaces, header uppercase inconsistency, removed disqus, fixed index.rst

erazor666 committed:
Fixed some travis complaints

erazor666 committed:
More travis work

erazor666 committed:
Travis fixes

erazor666 committed:
Update oled.rst

erazor666 committed:
Update oled.rst

erazor666 committed:
Update oled.rst

erazor666 committed:
Update oled.rst

erazor666 commented:
@OttoWinter any idea about this?

```
1:21:06 AM: Warning, treated as error:
1:21:06 AM: /opt/build/repo/cookbook/oled.rst:43:Error in "code-block" directive:
1:21:06 AM: 1 argument(s) required, 0 supplied.
1:21:06 AM: .. code-block::
1:21:06 AM:     [11:54:53][C][i2c:043]: I2C Bus:
1:21:06 AM:     [11:54:53][C][i2c:044]:   SDA Pin: GPIO14
1:21:06 AM:     [11:54:53][C][i2c:045]:   SCL Pin: GPIO12
1:21:06 AM:     [11:54:53][C][i2c:046]:   Frequency: 50000 Hz
1:21:06 AM:     [11:54:53][I][i2c:048]: Scanning i2c bus for active devices...
1:21:06 AM:     [11:54:53][I][i2c:055]: Found i2c device at address 0x3C
1:21:06 AM: make: *** [html] Error 2
```

OttoWinter commented on cookbook/oled.rst:
```diff
@@ -0,0 +1,273 @@
+    :align: center
+    :width: 80.0%
+
+The XY orientation of the display.
```
Indent with 4 spaces.

OttoWinter commented on cookbook/oled.rst:
```diff
@@ -0,0 +1,273 @@
+The XY orientation of the display.
+
+Configuration
+_____________
```
No two headers right next to each other. Add a short text below "configuration" if you want to keep this heading.

OttoWinter commented on cookbook/oled.rst:
```diff
@@ -0,0 +1,273 @@
+
+Configuration
+_____________
+i2c:
```
Make this longer. "i2c:" alone is too little descriptive. Something like I2C Bus Configuration

OttoWinter commented on cookbook/oled.rst:
```diff
@@ -0,0 +1,273 @@
+i2c:
+****
+
+If you don't know the i2c address for your display:
```
That is already the default.

OttoWinter replied:
Explain how the display is interfaced, and that this interface needs addresses to communicate. Not only "If you don't know the i2c address for your display:"

OttoWinter commented on cookbook/oled.rst:
```diff
@@ -0,0 +1,273 @@
+Take note of the address and set ``scan: false`` again.
+(The board used in this example is a Wemos D1 mini, so adapt pins to the board you are using)
+
+Node config:
```
Titlecase section headings

OttoWinter commented on cookbook/oled.rst:
```diff
@@ -0,0 +1,273 @@
+************
+
+The configuration for your node should look something like this:
+(your basic node configuration, e.g. network and such is not discussed in this guide):
```
Reverse this, first everything should be roughly explained, and at the end the full config will follow.

OttoWinter commented on index.rst:
```diff
@@ -295,6 +295,7 @@ Cookbook
     Mirabella Genio Bulb, cookbook/mirabella-genio-bulb, cookbook-mirabella-genio-b22-rgbw.jpg
     Garage Door, cookbook/garage-door, window-open.svg
     Brilliant / Mirabella Genio Smart Plugs, cookbook/brilliant-mirabella-genio-smart-plugs, cookbook-brilliant-mirabella-genio-smart-plugs.jpg
+    SSD1306 OLED Guide, cookbook/oled, oled.png
```
This should not be a PNG (png uses way too much space for these types of images). Make it a JPG.

OttoWinter commented on cookbook/oled.rst:
```diff
@@ -0,0 +1,273 @@
+- The waterdrop with a % sign inside it next to humidity sensor reading
+
+Images:
+_______
```
Headings are done with dash, not underline

OttoWinter commented on cookbook/oled.rst:
```diff
@@ -0,0 +1,273 @@
+    [11:54:53][I][i2c:055]: Found i2c device at address 0x3C
+
+Take note of the address and set ``scan: false`` again.
+(The board used in this example is a Wemos D1 mini, so adapt pins to the board you are using)
```
The user doesn't need to set scan to false again.

OttoWinter reviewed (commented):
- The cookbook images need to be in the images subfolder.

erazor666 committed:
placed in wrong dir

erazor666 committed:
placed in wrong dir

erazor666 committed:
placed in wrong dir

erazor666 committed:
change format

erazor666 committed:
Add files via upload

erazor666 committed:
change to jpg

erazor666 committed:
Made various corrections.

erazor666 committed:
Update oled.rst

erazor666 committed:
Update oled.rst

erazor666 committed:
Update oled.rst

erazor666 commented:
I think i got it all corrected now, the deploy/netlify check fails on something thats not related to this document i _think_

```
2:37:36 PM: reading sources... [ 12%] components/cover/template
2:37:36 PM: reading sources... [ 13%] components/debug
2:37:36 PM: reading sources... [ 13%] components/deep_sleep
2:37:36 PM: reading sources... [ 14%] components/display/index
2:37:36 PM: reading sources... [ 14%] components/display/lcd_display
2:37:36 PM: reading sources... [ 15%] components/display/max7219
2:37:37 PM: reading sources... [ 15%] components/display/nextion
2:37:37 PM: reading sources... [ 16%] components/display/ssd1306
2:37:37 PM: reading sources... [ 17%] components/display/waveshare_epaper
2:37:37 PM: reading sources... [ 17%] components/esp32_ble_beacon
2:37:37 PM: reading sources... [ 18%] components/esp32_ble_tracker
2:37:37 PM: reading sources... [ 18%] components/esp32_camera
2:37:37 PM: reading sources... [ 19%] components/esphome
2:37:37 PM: reading sources... [ 19%] components/ethernet
2:37:46 PM: failed during stage 'building site': Build script returned non-zero exit code: 2
2:37:37 PM: reading sources... [ 20%] components/fan/binary
2:37:37 PM: reading sources... [ 20%] components/fan/index
2:37:47 PM: Shutting down logging, 161 messages pending
```

erazor666 committed:
Update oled.rst

OttoWinter committed:
Update oled.rst

OttoWinter commented:
@erazor666 Please see the comments I wrote before. Some of them are not addressed yet. I fixed some of it now myself but some are still left.

erazor666 committed:
Update oled.rst

erazor666 commented:
4 space indentation, fixed
2 headers, fixed
i2c added some more words, fixed
Removed unnecessary text regarding i2c scan, fixed
titlecase heading, fixed
reversed 2 lines, fixed
Headings underline, fixed
Moved images to correct folder, fixed
There was a line with over 160 chars that travis-ci didn't like.

Not sure on why the netlify check "crashes"

I may have missed something, but not sure what(?)
